README.adoc

```diff
@@ -1,50 +1,111 @@
-
-
-Please write commit messages using the following format:
-
-	AT-<issue number> <commit message>
+= agritrader
+
+This project is a mobile application to be used by traders that interact with farmers to collect agricultural goods.
+
+== Development
+
+=== Setting up a Development Environment via Docker
+
+A Docker development environment has been made available, to use it follow the steps below.
+
+IMPORTANT: This does not work on native docker for Windows and OS X.
+This works natively on Linux.
+It can work on Docker Toolbox by link:https://gist.github.com/stonehippo/e33750f185806924f1254349ea1a4e68[passing the USB device through to VirtualBox] first.
+
+. Install Docker using the instructions on the link:https://docs.docker.com/engine/installation/[official website].
+Do not use a package manager as you will get an outdated version of Docker.
+
+. Give executable permissions to the utility script using `chmod +x dockerize.sh`.
+
+. Build the image using `./dockerize.sh init`.
+This may take a while.
+
+. Install the node modules using `./dockerize.sh install`.
+
+. Plug in your phone to your computer and allow access to your phone.
+Allowing access to your phone varies on the phone but is commonly a pop-up shortly after plugging in a phone to a computer.
+Ensure the cable connecting the phone and computer is capable of data transfer and not just power.
+
+. Identify the path to your phone.
+This can be done using `lsusb`.
+For example the output could be:
++
+[source,bash]
+----
+user@user-pc ~/boresha/agritrader $ lsusb
+Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
+Bus 003 Device 003: ID 093a:2521 Pixart Imaging, Inc. Optical Mouse
+Bus 003 Device 002: ID 0534:0002 Motorola, Inc.
+Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
+Bus 001 Device 055: ID 04e8:6860 Samsung Electronics Co., Ltd Galaxy (MTP)
+Bus 001 Device 002: ID 046d:0a03 Logitech, Inc. Logitech USB Microphone
+Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+----
++
+The path would then be `/dev/bus/usb/001/055` for the Samsung device (phone).
+
+. Check that the Docker image can find your phone using `./dockerize.sh adb /path/to/device/`.
+The output should include a device.
++
+[source,bash]
+----
+user@user-pc ~/boresha/agritrader $ ./dockerize.sh adb /dev/bus/usb/001/055
+List of devices attached
+* daemon not running. starting it now at tcp:5037 *
+* daemon started successfully *
+ce12160cdb9b2f1905	device
+----
+
+. Build the package and deploy it to your phone using `./dockerize.sh android /path/to/device`.
+This should take a few minutes before anything shows up on your phone.
+
+. Once the app is running on your phone, enable hot reloading by shaking your phone and selecting _Enable Hot Reloading_.
+
+NOTE: The phone will need to be disconnected and reconnected for every build (`./dockerize.sh android /path/to/device`).
```

Comment:
Do you know why this is? Why does it have to be disconnected and reconnected every time?

Reply:
I think, *think*, when the adb daemon connects to your phone, your phone is aware of it, so it's a 2 way connection. The way docker ends itself doesn't properly clean up adb and so your phone still thinks it's connected to the old adb daemon so it can't connect to a new one. Unplugging it let's your phone know to disconnect from the old adb daemon and accept a new one.

Reply:
I'm not completely sure, I don't have a good grasp on exactly why this is happening, so there might be a really easy way to circumvent it that I don't know of. Overall this isn't so bad though since hot reloading works.